src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.AddSub.cs

```diff
@@ -68,5 +128,208 @@ internal static partial class BigIntegerCalculator
+
+            int leftLength = Length(left);
+            int rightLength = Length(right);
+
+            uint[] bits = new uint[leftLength];
+
+            fixed (uint* l = left, r = right, b = bits)
+            {
+                Subtract(l, leftLength,
+                         r, rightLength,
+                         b, bits.Length);
+            }
+
+            return bits;
+        }
+
+        [SecuritySafeCritical]
+        private unsafe static void Subtract(uint* left, int leftLength,
+                                            uint* right, int rightLength,
+                                            uint* bits, int bitsLength)
+        {
+            Debug.Assert(leftLength >= 0);
+            Debug.Assert(rightLength >= 0);
+            Debug.Assert(leftLength >= rightLength);
+            Debug.Assert(Compare(left, leftLength, right, rightLength) >= 0);
+            Debug.Assert(bitsLength == leftLength);
+
+            // Executes the "grammar-school" algorithm for computing z = a - b.
+            // While calculating z_i = a_i - b_i we take care of overflow:
+            // Since a_i - b_i doesn't need any additional bit, our carry c
+            // has always the value -1 or 0; hence, we're safe here.
+
+            int i = 0;
+            long carry = 0L;
+
+            for (; i < rightLength; i++)
+            {
+                long digit = (left[i] + carry) - right[i];
+                bits[i] = (uint)digit;
+                carry = digit >> 32;
+            }
+            for (; i < leftLength; i++)
+            {
+                long digit = left[i] + carry;
+                bits[i] = (uint)digit;
+                carry = digit >> 32;
+            }
+            bits[i] = (uint)carry;
+        }
+
+        [SecuritySafeCritical]
+        private unsafe static void SubtractSelf(uint* left, int leftLength,
+                                                uint* right, int rightLength)
+        {
+            Debug.Assert(leftLength >= 0);
+            Debug.Assert(rightLength >= 0);
+            Debug.Assert(leftLength >= rightLength);
+            Debug.Assert(Compare(left, leftLength, right, rightLength) >= 0);
+
+            // Executes the "grammar-school" algorithm for computing z = a - b.
+            // Same as above, but we're writing the result directly to a and
+            // stop execution, if we're out of b and c is already 0.
+
+            int i = 0;
+            long carry = 0L;
+
+            for (; i < rightLength; i++)
+            {
+                long digit = (left[i] + carry) - right[i];
+                left[i] = (uint)digit;
+                carry = digit >> 32;
+            }
+            for (; carry != 0 && i < leftLength; i++)
+            {
+                long digit = left[i] + carry;
+                left[i] = (uint)digit;
+                carry = digit >> 32;
+            }
+        }
+
+        public static int Compare(uint[] left, uint[] right)
```

Comment:
Theoretically, yes. But it would involve additional "fixing" of these arrays, and I wanted to avoid that. Thus, we have one version for ordinary arrays and one for pointers.